### PR TITLE
Instead of rethrowing errors, log them. Fixes gh-449

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: bin/tessel-2.js [hunk 1]
```diff
@@ -287,18 +287,27 @@ module.exports.closeSuccessfulCommand = function() {
...
-module.exports.closeFailedCommand = function(opts, err) {
-  if (!err) {
-    err = opts;
-    opts = {};
-  }
-  if (err instanceof Error) {
-    throw err;
+module.exports.closeFailedCommand = function(opts, error) {
+
+  if (opts) {
```
Comment (HipsterBrown) [inline bin/tessel-2.js]: Is there a case where this would ever get called without at least `opts` passed in?

**Comment by HipsterBrown** [comment]
Besides that one comment, it looks good to me. It will be nice to have some better error feedback from every command.

File: bin/tessel-2.js (hunk 1 above)
Comment (rwaldron) [inline bin/tessel-2.js]: There were tests in `test/unit/bin-tessel-2.js` that broke when I assumed that couldn't happen. I'm going to investigate.

Comment (rwaldron) [inline bin/tessel-2.js]: I found no occurrences where `closeFailedCommand` is called with nothing, so I'm re-working this. I was also able to clean up the syntax error handling from https://github.com/tessel/t2-cli/commit/0726002569bb3d043e0b22edb2a5ac51e86b304c

Comment (HipsterBrown) [inline bin/tessel-2.js]: Awesome! Simpler code is happier code. :smile:

File: bin/tessel-2.js [hunk 2]
```diff
@@ -40,12 +40,7 @@ function makeCommand(commandName) {
 
 function callControllerWith(methodName, opts) {
   return controller[methodName](opts)
-    .then(module.exports.closeSuccessfulCommand, function(opts, err) {
-      if (typeof opts === 'object' && opts !== null) {
-        err = opts.message || '';
-      }
-      module.exports.closeFailedCommand(opts, err);
-    });
+    .then(module.exports.closeSuccessfulCommand, module.exports.closeFailedCommand);
```
Comment (HipsterBrown) [inline bin/tessel-2.js]: This is so much more approachable to follow. :clap:

**Comment by HipsterBrown** [comment]
Fantastic refactor. :+1:

**Comment by rwaldron** [comment]
@HipsterBrown thanks for the review (x2)

**Pull request merged**